daemon/algod/api/server/router.go

```diff
@@ -117,12 +118,21 @@ func NewRouter(logger logging.Logger, node *node.AlgorandFullNode, shutdown <-ch
 	// Request Context
 	ctx := lib.ReqContext{Node: node, Log: logger, Shutdown: shutdown}
 
-	// Registers /debug/pprof handler under root path and under /urlAuth path
-	// to support header or url-provided token.
-	router.PathPrefix("/debug/pprof/").Handler(http.DefaultServeMux)
-
-	urlAuthRouter := router.PathPrefix(urlAuthEndpointPrefix)
-	urlAuthRouter.PathPrefix("/debug/pprof/").Handler(http.DefaultServeMux).Name(debugRouteName)
+	// Route pprof requests
+	if node.Config().EnableProfiler {
+		// Registers /debug/pprof handler under root path and under /urlAuth path
+		// to support header or url-provided token.
+		router.PathPrefix(pprofEndpointPrefix).Handler(http.DefaultServeMux)
+
+		urlAuthRouter := router.PathPrefix(urlAuthEndpointPrefix)
+		urlAuthRouter.PathPrefix(pprofEndpointPrefix).Handler(http.DefaultServeMux).Name(debugRouteName)
+	} else {
+		// pprof disabled, route to black hole
+		disabledHandler := func(w http.ResponseWriter, r *http.Request) {
+			http.Error(w, "Profiler disabled", http.StatusForbidden)
```

Comment:
I think that the desired behavior here is to return the same output as if "there is nothing there" while logging the incident.

Reply:
Sure, we can just not route pprof requests at all, done.